WordPress/src/test/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailViewModelTest.kt

```diff
@@ -1,25 +1,157 @@
+    private val getCategoriesUseCase: GetCategoriesUseCase = mock()
+    private val addCategoryUseCase: AddCategoryUseCase = mock()
+    private val resourceProvider: ResourceProvider = mock()
+    private val selectedSiteRepository: SelectedSiteRepository = mock()
 
-    private lateinit var viewModel: CategoryDetailViewModel
+    private val uiStates = mutableListOf<UiState>()
+    private val onCategoryPushStates = mutableListOf<Event<CategoryUpdateUiState>>()
+
+    private val siteModel: SiteModel = mock()
+    private val topLevelCategory = "Top level"
 
     @Before
     fun setUp() {
-        viewModel = CategoryDetailViewModel(TEST_DISPATCHER, dispatcher)
+        setUpMockResponse()
+        viewModel = CategoryDetailViewModel(
+                TEST_DISPATCHER,
+                networkUtilsWrapper,
+                getCategoriesUseCase,
+                addCategoryUseCase,
+                resourceProvider,
+                dispatcher,
+                selectedSiteRepository
+        )
+        setUpUiStateObservers()
+    }
+
+    private fun setUpUiStateObservers() {
+        viewModel.uiState.observeForever { if (it != null) uiStates += it }
+        viewModel.onCategoryPush.observeForever { if (it != null) onCategoryPushStates += it }
+    }
+
+    private fun setUpMockResponse() {
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(siteModel)
+        whenever(resourceProvider.getString(R.string.top_level_category_name)).thenReturn(topLevelCategory)
+        whenever(getCategoriesUseCase.getSiteCategories(siteModel)).thenReturn(arrayListOf())
+    }
+
+    @Test
+    fun `when vm starts, then default parent category is selected`() {
+        viewModel.start()
+        assertEquals(topLevelCategory, uiStates.first().categories[0].name)
+    }
+
+    @Test
+    fun `when vm starts, then submit button is shown and disabled`() {
+        viewModel.start()
+        assertFalse(uiStates.first().submitButtonUiState.enabled)
+        assertTrue(uiStates.first().submitButtonUiState.visibility)
+    }
+
+    @Test
+    fun `given category name is updated, when vm starts, then submit button is enabled`() {
+        viewModel.start()
+        viewModel.onCategoryNameUpdated("category name")
+        assertTrue(uiStates.last().submitButtonUiState.enabled)
     }
 
     @Test
-    fun `sample test`() {
-    } // TODO:
+    fun ` given parent category is updated, when vm starts, then ui state is updated`() {
+        val categoryNode = CategoryNode(1, 1, "Category")
+        whenever(getCategoriesUseCase.getSiteCategories(siteModel)).thenReturn(arrayListOf(categoryNode))
+        viewModel.start()
+        val selectedCategoryParent = 2
+        viewModel.onParentCategorySelected(selectedCategoryParent)
+        assertEquals(selectedCategoryParent, uiStates.last().selectedParentCategoryPosition)
+    }
+
+    @Test
+    fun `given no internet, when submit is invoked, then ui state is updated`() {
```

Comment:
`then` clause description looks misleading, can you recheck?

Reply:
Yup, Thanks @ashiagr. 🙌 .The test statement was incorrect. Corrected in 1c5047c